pretext/ExtraTopics/Glossary.ptx

```diff
@@ -0,0 +1,156 @@
+    </gi>
+    <gi>
+      <title>algorithm</title>
+      <p>A general step by step process for solving a problem.</p>
+    </gi>
+    <gi>
+      <title>bug</title>
+      <p>An error in a program.</p>
+    </gi>
+    <gi>
+      <title>byte code</title>
+      <p>An intermediate language between source code and object code. Many
+                            modern languages first compile source code into byte code and then
+                            interpret the byte code with a program called a <em>virtual machine</em>.</p>
+    </gi>
+    <gi>
+      <title>codelens</title>
+      <p>An interactive environment that allows the user to control the step by step execution of a Python program</p>
+    </gi>
+    <gi>
+      <title>comment</title>
+      <p>Information in a program that is meant for other programmers (or anyone
+                            reading the source code) and has no effect on the execution of the
+                            program.</p>
+    </gi>
+    <gi>
+      <title>compile</title>
+      <p>To translate a program written in a high-level language into a
+                            low-level language all at once, in preparation for later execution.</p>
+    </gi>
+    <gi>
+      <title>debugging</title>
+      <p>The process of finding and removing any of the three kinds of
+                            programming errors.</p>
+    </gi>
+    <gi>
+      <title>exception</title>
+      <p>Another name for a runtime error.</p>
+    </gi>
+    <gi>
+      <title>executable</title>
+      <p>Another name for object code that is ready to be executed.</p>
+    </gi>
+    <gi>
+      <title>formal language</title>
+      <p>Any one of the languages that people have designed for specific
+                            purposes, such as representing mathematical ideas or computer programs;
+                            all programming languages are formal languages.</p>
+    </gi>
+    <gi>
+      <title>high-level language</title>
+      <p>A programming language like Python that is designed to be easy for
+                            humans to read and write.</p>
+    </gi>
+    <gi>
+      <title>interpret</title>
+      <p>To execute a program in a high-level language by translating it one
+                            line at a time.</p>
+    </gi>
+    <gi>
+      <title>low-level language</title>
+      <p>A programming language that is designed to be easy for a computer to
+                            execute; also called machine language or assembly language.</p>
+    </gi>
+    <gi>
+      <title>natural language</title>
+      <p>Any one of the languages that people speak that evolved naturally.</p>
+    </gi>
+    <gi>
+      <title>object code</title>
+      <p>The output of the compiler after it translates the program.</p>
+    </gi>
+    <gi>
+      <title>parse</title>
+      <p>To examine a program and analyze the syntactic structure.</p>
```

Comment:
I mean thats half true. It gets parsed to a form that the next step in the language can understand it. In python, it gets parsed to the AST.